CMakeLists.txt

```diff
@@ -18,8 +18,8 @@ if(WIN32)
     include_directories(${ASIO_HEADER_PATH})
 endif()
 
-OPTION(BUILD_EXAMPLES "Build Library with examples" OFF)
-OPTION(BUILD_TESTS "Build Library with tests" OFF)
+OPTION(BUILD_EXAMPLES "Build Library with examples" ON)
+OPTION(BUILD_TESTS "Build Library with tests" ON)
```

Comment:
The test can stay optional, i.e. keep `BUILD_TESTS` as it was before. I just wanted to have the examples build by default to directly see compilation issues with e.g. templates.